refactor(Purchase Order) make link readable

lomamech opened the pull request:
- Update purchase order registry make all search Readable
- Use checkbox instead radio for the research On Status
- Update research in Purchase order Registry, make link Readable
- Add the new table purchase_status
- Update schema for table purchase, add new property status_id
- Remove the property is_confirme, is_received, is_partially_received,
  is_cancelled from property of Purchase by the using of status_id

this issue reference #2139 and #2130

lomamech committed:
refactor(Purchase Order)
- Update purchase order registry make all search Readable
- Use checkbox instead radio for the research On Status

lomamech committed:
refactor(Purchase Order Registry)
- Update research in Purchase order Registry, make link Readable
- Add the new table purchase_status
- Update schema for table purchase, add new property status_id
- Remove the property is_confirme, is_received, is_partially_received,
  is_cancelled from property of Purchase by the using of status_id

jniles commented on client/src/modules/purchases/create/create.html:
```diff
@@ -51,7 +51,7 @@
                     <div class="radio">
                       <label>
                         <input type="radio" name="paymentMethod" id="on-delivery" value="FORM.LABELS.ON_DELIVERY" ng-model="PurchaseCtrl.order.details.payment_method" required>
-                        <sapn translate>FORM.LABELS.ON_DELIVERY</span>>
```
:dancer:

jniles commented on client/src/modules/purchases/templates/cellStatus.tmpl.html:
```diff
@@ -1,17 +1,7 @@
-    <div ng-if="row.entity.is_cancelled === 1" class="label label-danger" translate>
-       PURCHASES.STATUS.CANCELLED
-    </div>
+<div class="ui-grid-cell-contents">        
```
Much cleaner! :+1:

jniles commented on server/controllers/finance/reports/purchases/report.handlebars:
```diff
@@ -39,35 +39,13 @@
-                </strong>
-              </td>
+              <td>{{currency cost currency_id}}</td>
+              <td><strong>{{translate status}} </strong></td>
```
Much cleaner! :+1:

jniles commented on client/src/modules/purchases/modals/status.js:
```diff
@@ -6,30 +6,29 @@ PurchaseOrderStatusModalController.$inject = [
-        updateOrder(vm.purchase.uuid, data);
-    }
+  // submit the choice 
+  function submit(status) {
```
Is status ever used here?

jniles commented on client/src/modules/purchases/modals/status.js:
```diff
@@ -6,30 +6,29 @@ PurchaseOrderStatusModalController.$inject = [
+  // submit the choice 
+  function submit(status) {
+    var data = { status_id : vm.status };
+    updateOrder(vm.purchase.uuid, data);
```
If you `return updateOrder()` here and `return PurchaseOrder.update()` on line 28, you will get a loading indicator on the modal's submit button.

jniles commented on client/src/modules/purchases/modals/search.tmpl.html:
```diff
@@ -43,26 +43,10 @@
-              <span translate>PURCHASES.STATUS.CANCELLED</span>
-            </label>
-          </div>
+          <bh-purchase-status-select
```
Nice :+1:

jniles reviewed (commented):
Looks pretty good!  I've made a few comments below.

@mbayopanda, could you take a look at this?  You did the original implementation ... does this still meet your expectations?

lomamech committed:
Resolve Conflict

lomamech committed:
refactor(Purchase Order Status Controller)
- Remove not necessary param in function Submit
- remove not necessary function updateOrder

jniles reviewed (approved):
@lomamech, I think @mbayopanda is pretty busy on another project.  The code LGTM.  We can modify it later if necessary.

bors r+
